streams/src/main/java/org/apache/kafka/streams/state/internals/QueryableStoreProvider.java

```diff
@@ -23,17 +23,15 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.util.Collections.singletonList;
-
 /**
  * A wrapper over all of the {@link StateStoreProvider}s in a Topology
  */
 public class QueryableStoreProvider {
 
-    private final List<StateStoreProvider> storeProviders;
+    private final List<StreamThreadStateStoreProvider> storeProviders;
```

Comment:
thinking aloud: even though we are binding to a specific StateStoreProvider implementation here, it seems fine,since there are n't any other really in a topology

Reply:
This is part of the point of this change, actually, `StreamThreadStateStoreProvider` is no longer a `StateStoreProvider`. If you need a StateStoreProvider, you have to adapt it with `WrappingStoreProvider`.